### PR TITLE
exception: device keystore

### PR DESCRIPTION
**Pull request by minh-rakuten**
# Description
Throw exception for SDK when cannot create or get crypto key.

## Links
MINI-2733

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/CachedMiniAppVerifier.kt [hunk 1]
```diff
@@ -77,3 +71,17 @@ internal class CachedMiniAppVerifier
...
+    EncryptedSharedPreferences.create(
+        "com.rakuten.tech.mobile.miniapp.cache.hash",
+        MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC),
+        context,
+        EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+        EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+    )
+} catch (e: Exception) {
+    throw MiniAppSdkException("Something wrong with device keystore." +
+            "MiniApp SDK cannot proceed due to local security validation. " + e.message)
```
Comment (khairul-alam-licon) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/CachedMiniAppVerifier.kt:86]: Normally, do we need to provide that much details? I think, we could write such as..

`throw MiniAppSdkException("Something went wrong, MiniApp SDK cannot proceed due to security validation: " + e.message)`

**Review by khairul-alam-licon — APPROVED** [review]
Looks good, I have left a minor concern. 
Also, do we need to write this in changelog? A separate PR will be fine too.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/CachedMiniAppVerifier.kt [hunk 2]
```diff
@@ -77,3 +71,17 @@ internal class CachedMiniAppVerifier
...
+private fun initEncryptedSharedPreference(context: Context) = try {
+    EncryptedSharedPreferences.create(
+        "com.rakuten.tech.mobile.miniapp.cache.hash",
+        MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC),
+        context,
+        EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+        EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+    )
+} catch (e: Exception) {
+    throw MiniAppSdkException("Something wrong with device keystore." +
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/CachedMiniAppVerifier.kt:85]: Maybe we should make a specific exception type for this which extends `MiniAppSdkException`? Maybe something like `MiniAppVerificationException`. What do you think?

**Review by corycaywood — APPROVED** [review]
LGTM 👍  Just one thought below.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/CachedMiniAppVerifier.kt (hunk 2 above)
Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/CachedMiniAppVerifier.kt]: updated.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/CachedMiniAppVerifier.kt (hunk 1 above)
Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/CachedMiniAppVerifier.kt]: updated.

**Comment by khairul-alam-licon** [comment]
Yeah, I think we merge it 👍

**Comment by khairul-alam-licon** [comment]
Yeah, I think we can merge it 👍

**Pull request merged**